src/kafka/inflight_activation_batcher.rs

```diff
@@ -50,13 +50,22 @@ impl Reducer for InflightActivationBatcher {
     async fn reduce(&mut self, t: Self::Input) -> Result<(), anyhow::Error> {
         let runtime_config = self.runtime_config_manager.read().await;
         let task_name = &t.activation.taskname;
-        if !runtime_config.drop_task_killswitch.contains(task_name) {
-            self.buffer.push(t);
-        } else {
+
+        if runtime_config.drop_task_killswitch.contains(task_name) {
             metrics::counter!("filter.drop_task_killswitch", "taskname" => task_name.clone())
                 .increment(1);
+            return Ok(());
         }
 
+        if let Some(expires_at) = t.expires_at {
```

Comment:
Can you add a unit test for this too please.

Reply:
Yes, added 👍